### PR TITLE
samples: nrf9160: modem_shell: add bootstrap support for LwM2M

### PR DESCRIPTION
**Pull request by tokangas**
Added a new configuration overlay file to enable bootstrapping with LwM2M.

### CODE REVIEW

File: samples/nrf9160/modem_shell/README.rst [hunk 1]
```diff
@@ -1140,13 +1140,17 @@ LwM2M support
 
 Before building and running the sample, select the LwM2M server for testing.
 Follow the instructions in :ref:`server_setup_lwm2m_client` to set up the server and register your device to the server.
+With the default LwM2M configuration, the device connects directly to the device management server without bootstrap support.
+You can change the LwM2M server address by setting :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_SERVER`.
```
Comment (peknis) [inline samples/nrf9160/modem_shell/README.rst:1144]:
```suggestion
You can change the LwM2M server address by setting the :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_SERVER` Kconfig option.
```

File: samples/nrf9160/modem_shell/README.rst [hunk 2]
```diff
@@ -1146,6 +1146,7 @@ As of now, you can only use AVSystem's Coiote LwM2M server for the location assi
 
 To build the MoSh sample with LwM2M support, use the ``-DOVERLAY_CONFIG=overlay-lwm2m.conf`` option and set the used Pre-Shared-Key (PSK) using :kconfig:option:`CONFIG_MOSH_LWM2M_PSK` Kconfig option.
 You can change the LwM2M server address by setting :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_SERVER`.
+To enable bootstrapping, use the optional overlay file :file:`overlay-lwm2m_bootstrap.conf`.
 To enable P-GPS support, use the optional overlay files :file:`overlay-lwm2m_pgps.conf` and :file:`overlay-pgps.conf`.
```
[the remark this answers is not in the document]
Comment (peknis) [reply samples/nrf9160/modem_shell/README.rst]: Very nice!

**Review by peknis — APPROVED** [review]

File: samples/nrf9160/modem_shell/sample.yaml [hunk 3]
```diff
@@ -77,6 +77,13 @@ tests:
       - nrf9160dk_nrf9160_ns
     platform_allow: nrf9160dk_nrf9160_ns
     tags: ci_build
+  sample.nrf9160.modem_shell.lwm2m_bootstrap:
+    build_only: true
+    extra_args: OVERLAY_CONFIG="overlay-lwm2m.conf;overlay-lwm2m_bootstrap.conf"
+    integration_platforms:
+      - nrf9160dk_nrf9160_ns
+    platform_allow: nrf9160dk_nrf9160_ns
+    tags: ci_build
```
Comment (trantanen) [inline samples/nrf9160/modem_shell/sample.yaml:86]: This is not directly related to this PR but because I didn't see CONFIG_MOSH_LWM2M_PSK set here, I started to check how it's handled. We discussed offline that CONFIG_MOSH_LWM2M_PSK would be set empty by default and BUILD_ASSERT would be made to avoid misconfiguration of an empty value. This means user should always set it when compiling.

Comment (tokangas) [reply samples/nrf9160/modem_shell/sample.yaml]: Pushed a new commit which removes the default key and adds a build assert if the key is not configured.

**Review by jhirsi — APPROVED** [review]

**Review by trantanen — APPROVED** [review]

**Review by hiltunent — APPROVED** [review]
